### ISSUE TITLE
Audioplayer scan fails with LDAP integration

### ISSUE THREAD

**Issue by wizdude** [Rello/audioplayer#138]
### Expected behavior

scanning for audio files should scan and index audio files.

### Actual behavior

Indexing of audio tracks does not work. browser sits at "Start scanning" forever.

Trying the same from the command line reveals the actual cause of the problem:

root@nextcloud:/var/www# sudo -u apache php /var/www/htdocs/occ audioplayer:scan wiz
User wiz does not exist

by checking the data directory, i see (as expected) my home folder is actually my LDAP ID (derived from my active directory UID). if i use this on the command line it all works as expected. (UID below has been modified)

root@nextcloud:/var/www# sudo -u apache php /var/www/htdocs/occ audioplayer:scan A69BE269-1061-4182-A464-696969696969
Start scan for A69BE269-1061-4182-A464-696969696969
Audios found: 157
Duplicates found: 0
Written to music library: 157
Albums found: 3
Errors: 0

the audioplayer code needs to be modified to work with the user id and not the login name.

### Steps to reproduce the behavior
1.  Click on "Scan for audio files". Stays stuck on "Start scanning".
or
2. Try to index from command line. This gives more diagnostic information.

### Possible solution (optional)

Code is assuming path to users folder is based on their login name. It needs to be based on the user id as derived from Nextcloud.

ie: my login name = wiz. Audioplay assumes my home folder is /var/www/data/wiz
nextcloud internally uses A69BE269-1061-4182-A464-696969696969. the actual path to my home folder is /var/www/data/A69BE269-1061-4182-A464-696969696969

### Server configuration
Operating system = Slackware64 14.2

Web Server = Apache 2.4.25

Database = MariaDB 10.0.29

PHP = 5.6.29

Nextcloud 11.0.1.2

Fresh install of Nextcloud.

Audio Player version 1.4.1

Installed Audio Player from Apps in Nextcloud

No external storage

No encryption

### Client configuration

Operating system: Windows 10

Browser: Firefox

#### ownCloud/Nextcloud log

No log entries are created while the attempted fresh/index is taking place.

**Comment by Rello** [comment]
@wizdude how is occ file:scan behaving in this case? I am referring to the official coding and could not find a way that LDAP is recognized there.

thank you for your input. I am looking at the web-scanner in parallel.

**Comment by wizdude** [comment]
"admin" is normal internal user. "wiz" is LDAP user. it performs the same way with occ files:scan as well. see below.

i don't know if this helps at all, but another issue i encountered earlier on was with "share link" e-mails generated by nextcloud. they were sending out with the UID in the e-mail instead of the persons name. this was discussed here:

https://github.com/nextcloud/server/pull/3080

and code changes here:

https://github.com/nextcloud/server/pull/3080/files

let me know if i can be of any assistance testing anything. i'm keen to see this bug squashed since the audioplayer is essentially crippled in my nextcloud instance.


```
root@nextcloud:/var/www/htdocs# sudo -u apache php occ files:scan admin
Starting scan for user 1 out of 1 (admin)

+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 12      | 20    | 00:00:00     |
+---------+-------+--------------+
root@nextcloud:/var/www/htdocs# sudo -u apache php occ files:scan wiz
Unknown user 1 wiz

+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 0       | 0     | 00:00:00     |
+---------+-------+--------------+
root@nextcloud:/var/www/htdocs# sudo -u apache php occ files:scan A27BE269-1061-4182-A464-696969696969
Starting scan for user 1 out of 1 (A27BE269-1061-4182-A464-696969696969)

+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 61      | 272   | 00:00:01     |
+---------+-------+--------------+

```

**Comment by Rello** [comment]
Hello,
thank you for the input.
the problem is that file:scan is working the same logic as AP. there you also could not scan with "wiz".
the link you mentioned is deriving the display name from a UID. your UID is the LDAP number. your display name would be be a "first name, last name" if you maintained it, if I understood correctly.

I still have a question in the forum open also - lets see if someone has an answer there.

but: when you are working with the AP, the online scanner from the web-app should work. correct?
because there it should derive the corresponding userfolder from the login-session. if not, can you please check if you have an entry in the logfile after that?

sorry for not giving a better answer. I could not find anything in the documentation and also don´t have a chance to debug because I am not using LDAP.

...

**Comment by wizdude** [comment]
greetings,

for the life of me i can't reproduce this error. even since i did a scan using occ from the command line, it all appears to be working correctly.

i even created a new user, logged in, uploaded some music and it's all working fine.

is it possible that the very first scan from the web browser might be missing something (a database table or something) which gets created when you use the command line?

i do receive this error message in the logs, but it doesn't appear to affect the operation of AP. 

`{"reqId":"Fjbw71Np3wy2gHjSLJnw","remoteAddr":"124.69.69.69","app":"PHP","message":"fwrite() expects parameter 2 to be string, array given at \/var\/www\/htdocs\/lib\/private\/Files\/Storage\/Wrapper\/Encryption.php#230","level":3,"time":"2017-02-26T12:13:38+00:00","method":"POST","url":"\/apps\/audioplayer\/scanforaudiofiles","user":"admin","version":"11.0.1.2"}
`

if i perform a new fresh install of nextcloud for someone else and i'm able to reproduce this bug i'll certainly come back. in the mean time, it's all working well.

keep up the good work! :-)

**Comment by Rello** [comment]
really?
ok, even better.

this error I saw several times already. its not directly caused by AP - but I think somehow indirectly triggered.
as the scanner will be improved, I hope to get rid of this.

...I will close this for now...
